distributed/scheduler.py

```diff
@@ -7702,7 +7698,7 @@ def profile_to_figure(state):
         # Task stream
         task_stream = self.get_task_stream(start=start)
         total_tasks = len(task_stream)
-        timespent = defaultdict(int)
+        timespent: dict[int, float] = defaultdict(int)
```

Comment:
```suggestion
        timespent: "defaultdict[int, float]" = defaultdict(int)
```

Reply:
looks like we're abusing defaultdict while we should really use a Counter?

Reply:
sorry, I think my annotation is just wrong and it shouldn't be initialized to int but rather float, if we're 100% accurate

`timespent: defaultdict[str, float] = defaultdict(float)`

Does mypy recognize defaultdict?

Reply:
mypy recognizes everything in collections and collections.abc as long as you have ``from __future__ import annotations`` in your module

Reply:
still, I think you should use a Counter?

Reply:
> still, I think you should use a Counter?

Am I missing something obvious? How is this a counter? start, stop are floats / `time.time()`, we're not counting
`timespent[x["action"]] += x["stop"] - x["start"]`

Reply:
Counter and defaultdict(int) are pretty much the same thing, they only differ in how you initialise them. Nevermind, it's really a philosophical difference.